File: data/EXP01/python_3/2135/2135.output_2.py
# @lc app=leetcode id=2135 lang=python
class Solution:
    def wordCount(self, startWords, targetWords):
        count = 0   # to keep track of the number of strings that can be formed
        
        for target in targetWords:
            for start in startWords:
                start_set = set(start)
                target_set = set(target)
                
                if target_set.issubset(start_set):
                    count += 1   # increment count if it is possible to obtain target from start
                    break
                    
        return count

File: data/EXP01/golang_no_system_var/2135/2135.output_2.go
func wordCount(startWords []string, targetWords []string) int {
    count := 0
    for _, target := range targetWords {
        for _, start := range startWords {
            if len(start)-len(target) == 0 {
                startRunes := []rune(start)
                targetRunes := []rune(target)
                sort.Slice(startRunes, func(i, j int) bool { return startRunes[i] < startRunes[j] })
                sort.Slice(targetRunes, func(i, j int) bool { return targetRunes[i] < targetRunes[j] })
                if string(startRunes) == string(targetRunes) {
                    count++
                    break
                }
            } else if len(start)-len(target) == 1 {
                startRunes := []rune(start)
                targetRunes := []rune(target)
                foundLetters := make(map[rune]bool)
                for _, letter := range startRunes {
                    foundLetters[letter] = true
                }
                possible := true
                for _, letter := range targetRunes {
                    if !foundLetters[letter] {
                        possible = false
                        break
                    }
                }
                if possible {
                    count++
                    break
                }
            }
        }
    }
    return count
}
